src/Controls/src/Core/Page/Page.cs

```diff
@@ -606,20 +606,21 @@ internal virtual void OnChildMeasureInvalidated(VisualElement child, Invalidatio
 			InvalidateMeasureInternal(InvalidationTrigger.MeasureChanged);
 		}
 
+		EventHandler _onAppearingEventHandler;
 		internal void OnAppearing(Action action)
```

Comment:
@PureWeen not sure if it is ever the case where multiple callers will invoke this method before it appears and thus we need some sort of list of actions to run once it appears?

Maybe some list of actions is better anyway and a single event handler that just runs all the actions. And the disappearing just clears the list? Like a list of pending onappearing actions? For example, we have the `_pendingActions` list for the alerts and popups, so maybe we need to have a list of things that run once the page appears?